5.2.cup_of_join.py

```diff
@@ -0,0 +1,15 @@
+def join(*lists, sep="-"):
+    """
+    Concatenates any number of lists into a single list with each element separated by a specified separator string.
+    :param lists: Any number of lists to be concatenated.
+    :param sep: The separator string to be placed between each element of the concatenated list. Defaults to "-".
+    :return: A new list containing all of the elements from the input lists, separated by the specified
+    separator string.
+    """
+    if not lists:
+        return None
+    result_list = []
+    for lst in lists:
+        result_list += lst
+        result_list.append(sep)
+    return result_list[:-1]
```

Comment:
use name == main
https://www.tutorialspoint.com/What-does-the-if-name-main-do-in-Python